### PR TITLE
Phase2-hgx358C Add a few new modes to enable cog computation for wafers as default for the future V19 version of HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a few new modes to enable cog computation for wafers as default for the future V19 version of HGCal

#### PR validation:

Use the runTheMatrix test workflow

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by srimanob** [comment]
Note for the next PR:

```
src/SimCalorimetry/HGCalSimAlgos/src/HGCalSciNoiseMap.cc:167:10: warning: Value stored to 'edge' during its initialization is never read [deadcode.DeadStores]
  167 |   double edge(refEdge_);  //start with reference 3cm of edge
```

**Comment by bsunanda** [comment]
@nothingface0, @rvenditti, @syuvivida, @tjavaid Please approve this PR

**Comment by bsunanda** [comment]
@nothingface0, @rvenditti, @syuvivida, @tjavaid Please approve this PR

**Comment by bsunanda** [comment]
@tjavaid Please approve this PR

**Comment by tjavaid** [comment]
+1

**Comment by rappoccio** [comment]
+1

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 1]
```diff
@@ -233,7 +237,7 @@ GlobalPoint HGCalGeometry::getPosition(const DetId& detid, bool debug) const {
                                         << id.iCell2;
       }
       xy = m_topology.dddConstants().locateCell(
-          id.zSide, id.iLay, id.iSec1, id.iSec2, id.iCell1, id.iCell2, true, true, false, debug);
+          id.zSide, id.iLay, id.iSec1, id.iSec2, id.iCell1, id.iCell2, true, true, true, cog, debug);
```
Comment (fwyzard) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc:240]: @bsunanda was it intended to flip the `norot` value from `false` to `true`, while adding the `cog` parameter ?

Comment (felicepantaleo) [reply Geometry/HGCalGeometry/src/HGCalGeometry.cc:240]: Unfortunately the summary and comparisons of the test are not there any more. 
@bsunanda , did you have a look at the 4047 differences found?

**Pull request merged**